test/gas/lib/isGasExpenseCorrect.js

```diff
@@ -0,0 +1,42 @@
+const assert = require("assert");
+const to = require("../../../lib/utils/to.js");
+
+const isGasExpenseCorrect = async(expectedGasPrice, setGasPriceOnTransaction = false, web3, accounts) => {
+  const transferAmount = web3.utils.toBN(web3.utils.toWei("5", "finney"));
```

Comment:
why hard code the transfer amount?

Reply:
Also it seems a bit weird that we get our BN dependency via web3 - we should just `const BN = require("bn.js")` and `const transferAmount = new BN(...);`

Reply:
Good alternative. Never saw this as a big issue consider web3 already exists.

Reply:
The hardcoded transfer amount seems good enough for validating gas used. Not seeing how using a different transfer amount will affect gas price validation.

Reply:
BN is a separate library from web3, so it's weird that web3 provides a toBN function at all. Using a transitive dependency as a direct dependency is a code smell.

> Not seeing how using a variable transfer amount will affect gas price validation
This is moot because I don't think this helper should be changing state, but it's more to do with cohesion/improper encapsulation. In this case we're taking a test parameter that should be encoded explicitly in the test itself and making it implicit by hiding it within a test helper.

Reply:
Ended up using `number-to-bn` over bn.js for also handling stringed values.

Reply:
This file was renamed.